File: app/src/main/java/com/example/museumapplication/utils/Permission/PermissionHelper.java
package com.example.museumapplication.utils.Permission;

import android.app.Activity;
import android.content.Context;
import android.content.pm.PackageManager;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

/**
 * Permission Helper
 *
 * @since 2019-12-13
 */
public class PermissionHelper {
    private Fragment mFragment;
    private PermissionInterface mPermissionInterface;

    /**
     * PermissionHelper Construct
     *
     * @param fragment Fragment
     * @param permissionInterface PermissionInterface
     */
    public PermissionHelper(@NonNull Fragment fragment, @NonNull PermissionInterface permissionInterface) {
        mFragment = fragment;
        mPermissionInterface = permissionInterface;
    }

    /**
     * Request Permissions
     */
    public void requestPermissions() {
        String[] deniedPermissions =
                PermissionUtil.getDeniedPermissions(mFragment.getContext(), mPermissionInterface.getPermissions());
        if (deniedPermissions.length > 0) {
            PermissionUtil.requestPermissions(
                    mFragment, deniedPermissions, mPermissionInterface.getPermissionsRequestCode());
        } else {
            mPermissionInterface.requestPermissionsSuccess();
        }
    }

    /**
     * Request Permissions Result
     *
     * @param requestCode requestCode
     * @param permissions permissions
     * @param grantResults grantResults
     * @return true:Request Permissions success
     */
    public boolean requestPermissionsResult(
            int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        if (requestCode == mPermissionInterface.getPermissionsRequestCode()) {
            boolean isAllGranted = true;
            for (int result : grantResults) {
                if (result == PackageManager.PERMISSION_DENIED) {
                    isAllGranted = false;
                    break;
                }
            }
            if (isAllGranted) {
                mPermissionInterface.requestPermissionsSuccess();
            } else {
                mPermissionInterface.requestPermissionsFail();
            }
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/example/museumapplication/utils/permission/PermissionHelper.kt
/*
 * Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.example.museumapplication.utils.permission

import android.content.pm.PackageManager
import androidx.fragment.app.Fragment
import com.example.museumapplication.utils.permission.PermissionUtil.getDeniedPermissions
import com.example.museumapplication.utils.permission.PermissionUtil.requestPermissions


class PermissionHelper

(private val mFragment: Fragment, private val mPermissionInterface: PermissionInterface) {

    fun requestPermissions() {
        val deniedPermissions = getDeniedPermissions(mFragment.requireContext(), mPermissionInterface.permissions()!!)
        if (deniedPermissions.isNotEmpty()) {
            requestPermissions(
                    mFragment, deniedPermissions, mPermissionInterface.permissionsRequestCode())
        } else {
            mPermissionInterface.requestPermissionsSuccess()
        }
    }


    fun requestPermissionsResult(
            requestCode: Int, permissions: Array<String>, grantResults: IntArray): Boolean {
        if (requestCode == mPermissionInterface.permissionsRequestCode()) {
            var isAllGranted = true
            for (result in grantResults) {
                if (result == PackageManager.PERMISSION_DENIED) {
                    isAllGranted = false
                    break
                }
            }
            if (isAllGranted) {
                mPermissionInterface.requestPermissionsSuccess()
            } else {
                mPermissionInterface.requestPermissionsFail()
            }
            return true
        }
        return false
    }
}